src/main/java/gg/amy/catnip/utilities/FinderUtil.java

```
@@ -0,0 +1,413 @@
+                .collect(Collectors.toList());
+            if (!users.isEmpty())
+                return users;
+        } else if (DISCORD_ID.matcher(query).matches()) {
+            User user = catnip.cache().users().getById(query);
+            if (user != null)
+                return Collections.singletonList(user);
+        }
+
+        ArrayList<User> exact = new ArrayList<>();
+        ArrayList<User> wrongcase = new ArrayList<>();
+        ArrayList<User> startswith = new ArrayList<>();
+        ArrayList<User> contains = new ArrayList<>();
+        String lowerquery = query.toLowerCase();
+        catnip.cache().users().forEach(user -> {
+            String name = user.username();
+            if (name.equals(query))
+                exact.add(user);
+            else if (name.equalsIgnoreCase(query) && exact.isEmpty())
+                wrongcase.add(user);
+            else if (name.toLowerCase().startsWith(lowerquery) && wrongcase.isEmpty())
+                startswith.add(user);
+            else if (name.toLowerCase().contains(lowerquery) && startswith.isEmpty())
+                contains.add(user);
+        });
+        if (!exact.isEmpty())
+            return Collections.unmodifiableList(exact);
+        if (!wrongcase.isEmpty())
+            return Collections.unmodifiableList(wrongcase);
+        if (!startswith.isEmpty())
+            return Collections.unmodifiableList(startswith);
+        return Collections.unmodifiableList(contains);
+    }
+
+    public static List<User> findBannedUsers(String query, Guild guild) {
+        List<User> bans;
+
+        try {
+            bans = guild.catnip().rest().guild().getGuildBans(guild.id())
+                .toCompletableFuture().get().stream()
+                .map(GuildBan::user)
+                .collect(Collectors.toList());
+        } catch (Exception e) {
+            return null;
```

don't silently ignore exceptions